pkg/didcomm/protocol/didexchange/states.go

```diff
@@ -30,7 +36,7 @@ type state interface {
 	CanTransitionTo(next state) bool
 	// Executes this state, returning a followup state to be immediately executed as well.
 	// The 'noOp' state should be returned if the state has no followup.
-	Execute(msg dispatcher.DIDCommMsg) (followup state, err error)
+	Execute(msg dispatcher.DIDCommMsg, outbound dispatcher.Outbound, doc *did.Doc) (followup state, err error)
```

Comment:
Consider grouping both the outbound dispatcher and the `NewDidDoc()` function in a single interface - perhaps an internal `context` interface.

Reply:
okay will refactor like that